src/modules/navigator/mission.cpp

```diff
@@ -1925,3 +1928,127 @@ void Mission::publish_navigator_mission_item()
 
 	_navigator_mission_item_pub.publish(navigator_mission_item);
 }
+
+
+void Mission::calculate_mission_checksums()
+{
+	mission_s mission_state = {};
+	mission_stats_entry_s stats;
+	mission_checksum_s csum{};
+
+	csum.timestamp = hrt_absolute_time();
+	csum.mission_checksum = 0;
+	csum.fence_checksum = 0;
+	csum.rally_checksum = 0;
+	csum.all_checksum = 0;
+
+	dm_lock(DM_KEY_MISSION_STATE);
+
+	if (dm_read(DM_KEY_MISSION_STATE, 0, &mission_state, sizeof(mission_s)) != sizeof(mission_s)) {
+		PX4_ERR("dataman read failure");
+		return;
+	}
+
+	dm_unlock(DM_KEY_MISSION_STATE);
+
+	dm_item_t dm_current = (dm_item_t)(_mission.dataman_id);
+
+	for (int i = 0; i < mission_state.count; i++) {
+		struct mission_item_s mission_item = {};
+		const ssize_t len = sizeof(mission_item);
+
+		if (dm_read(dm_current, i, &mission_item, len) != len) {
+			PX4_ERR("dataman read failure");
```

Comment:
Same.

Reply:
I added consistent dm_lock / dm_unlock for every access of dm_item_t.